src/audio/copier/copier_ipcgtw.c

```diff
@@ -110,29 +109,26 @@ int copier_ipcgtw_process(const struct ipc4_ipcgtw_cmd *cmd,
 
 	buf = get_buffer(dev);
 
-	if (buf) {
-		buf_c = buffer_acquire(buf);
-	} else {
+	if (!buf) {
 		/* NOTE: this func is called from IPC processing task and can be potentially
 		 * called before pipeline start even before buffer has been attached. In such
 		 * case do not report error but return 0 bytes available for GET_DATA and
 		 * 0 bytes free for SET_DATA.
 		 */
-		buf_c = NULL;
 		comp_warn(dev, "copier_ipcgtw_process(): no buffer found");
 	}
```

Comment:
about " copier_ipcgtw_process() in copier_ipcgtw.c" I just meant this if block,  
I did not substitute buf for buf_c here, as we do not need to NULL it now